README.md

```diff
@@ -1,54 +1,190 @@
 # Sudo Sandwich is an example Heroku Add-on
 
-Example Heroku add-on built with Rails for the Add-on Partner API v3.
+Example Heroku add-on built with Rails for the v3 of the Platform API for
+Partners.
 
-Following instructions found at
+This add-on is meant to demonstrate integrations. Integrations can be written in
+any language or framework. We don't officially support this beyond that it's
+an example of how an add-on might be built.
+
+This add-on was built following instructions found at
 https://devcenter.heroku.com/articles/building-an-add-on.
 
-## Set up
+See [CONTRIBUTING.md](CONTRIBUTING.md) for information on how to set up and run
+the code for this application.
+
+## Add-on manifest
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#step-1-generate-your-add-on-manifest
+
+### Related code
+
+The manifest is not checked into version control because it contains secrets. A
+copy of the manifest with fake keys is located at
+[addon-manifest-example.json](addon-manifest-example.json). Although it is not
+checked into version control, it is helpful to know that the real manifest is
+named `addon-manifest.json` and will be called "the manifest" from this point
+forward.
+
+## Basic auth
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#basic-authentication
+
+### Related code
+
+Basic auth is implemented via the [`HttpBasicAuth`
+concern](app/controllers/concerns/http_basic_auth.rb). This concern is included
+in [`ApplicationController`](app/controllers/application_controller.rb), which
+all other controllers inherit from, because the Platform API for Partners sends
+basic auth credentials with all requests. Basic auth is selectively skipped for
+the single sign-on views, which are accessed by Heroku customers who are using
+the add-on rather than the Platform API for Partners.
+
+The username and password for basic auth are being accessed via environment
+variables whose values match the `slug` and `password` fields in the manifest.
+
+## The provisioning request
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#the-provisioning-request
```

Comment:
lol @ having four docs to link to - this is something to think about later

Reply:
ooh maybe we can split it into two sections - would it be two for async, two for sync?